### PR TITLE
Nullability annotations

### PR DESCRIPTION
**Pull request by chillpop**
- Added nullability annotations to Vokoder headers.
- updated Travis to use Xcode 7.1

Tested on both Xcode 7.1 and 6.4

@vokal/ios-developers, code review?

As before, I have still more changes to come.  I'm waiting to bump the version number and run pod update on the example projects until there are more changes.

### CODE REVIEW

**Comment by brockboland** [comment]
on it

**Comment by vokal-isaac** [comment]
Would it make sense to include something like
```objc
#if !__has_feature(nullability)
#define NS_ASSUME_NONNULL_BEGIN
#define NS_ASSUME_NONNULL_END
// … some other #defines so that the nullability things that don't exist are defined to nothing and won't cause compiler errors
#endif
```
so that this doesn't become unusable on pre-Xcode-6.4?

File: Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h [hunk 1]
```diff
@@ -70,7 +73,8 @@ typedef void(^VOKManagedObjectsReturnBlock)(NSArray *arrayOfManagedObjects);
  @param contextOfNil    The managed object context to update and/or insert the object. If nil, the main context will be used.
  @return                An instance of this subclass of NSManagedObject.
  **/
-+ (instancetype)vok_addWithDictionary:(NSDictionary *)inputDict forManagedObjectContext:(NSManagedObjectContext *)contextOrNil;
++ (nullable instancetype)vok_addWithDictionary:(nullable NSDictionary *)inputDict
```
Comment (designatednerd) [inline Pod/Classes/NSManagedObject+VOKManagedObjectAdditions.h]: Should the dictionary actually be nullable? If you're trying to import a nonexistent dictionary, that's just gonna fail anyway.

**Comment by designatednerd** [comment]
@vokal-isaac Do we have anyone still using 6.4?

**Comment by chillpop** [comment]
nullability was added in 6.3

**Comment by designatednerd** [comment]
@chillpop Didn't they change the way it works in 7 though?

**Comment by chillpop** [comment]
in 7, they changed the names to _Nullable and _Nonnull but __nullable and __nonnull are mapped to those in 7.

**Comment by brockboland** [comment]
Am I the only one that kind find details on the difference between `__Nullable` and `nullable` and `__nullable`?

**Comment by vokal-isaac** [comment]
@designatednerd I don't know if anyone's actively using 6.anything.  Maybe ask on Slack or check with lead(s) on any projects that started before Xcode 7 was released?  (If I had an in-flight tight-timeline project that started on 6.3 or 6.4, I'd keep it there until the timeline was met, unless something forced me off...)

**Comment by chillpop** [comment]
@brockboland (and others) this explains the details a bit: https://developer.apple.com/swift/blog/?id=25

**Comment by chillpop** [comment]
ELS is still on 6.4 but this will work back to 6.3.

**Comment by chillpop** [comment]
It's not too much of a change to support Xcode versions without nullability but it will make the property declarations uglier and possibly harder to scan for the nullability:
current
`@property (nullable, weak, readonly) UITableView *tableView;`
revised:
`@property (weak, readonly) UITableView * __nullable tableView;`
(with `#define __nullable` to make it a no-op)
I"m not sure it's worth supporting older versions of Xcode that we don't seem to have many projects (if any) actively using.

**Comment by vokal-isaac** [comment]
I'm not sure I particularly disagree about supporting older versions... but if we're going to break compatibility, I'd prefer to package up all the non-breaking changes into a patch or minor release, then put the breaking changes in a major release.

**Comment by seanwolter** [comment]
Wouldn't older Xcode users specify an older version of Vokoder? Oh. Isaac beat me to it. Great work!
:ghost: open-source ghost out! :ghost:

**Comment by chillpop** [comment]
In the interests of keeping this moving, I did what Isaac suggested.  Updates pushed.

**Comment by chillpop** [comment]
Additional updates pushed.  I realized that some headers don't include the main header so I had to put the degradation of nullability qualifiers in its own file.  This required a version bump and pod update on the sample projects to get them to include the new file.

**Comment by designatednerd** [comment]
Agreeing with @vokal-isaac that this should probably be packaged up with #58 for a major release, so you'll need to bump the podspec to that. But otherwise: LGTM.

**Comment by chillpop** [comment]
I've bumped the podspec to 1.4.0.  This should still be backwards compatible.

**Comment by designatednerd** [comment]
@chillpop: What version of Xcode is this backwards compatible to? We should probably note that in the README.

**Comment by chillpop** [comment]
Just checked: it builds and runs in Xcode 5.1

**Comment by designatednerd** [comment]
Wow. I'm impressed that you still HAVE Xcode 5.1. 

In that case: LGTM. @vokal-isaac @brockboland and oh why not @seanwolter anyone else have thoughts?

**Comment by chillpop** [comment]
I was surprised that I still had it too.

**Comment by chillpop** [comment]
updated README pushed.  Xcode 5 runs the example, but the tests use async testing methods that weren't introduced until Xcode 6.

**Comment by designatednerd** [comment]
SLGTM

**Comment by chillpop** [comment]
I'll wait until noon and then merge this.

**Pull request merged**